### PR TITLE
[Lora] correct lora saving & loading

### PR DESCRIPTION
**Pull request by patrickvonplaten**
Should correct: https://github.com/huggingface/diffusers/issues/2616

### CODE REVIEW

File: src/diffusers/loaders.py [hunk 1]
```diff
@@ -265,15 +262,6 @@ def save_function(weights, filename):
         # Save the model
         state_dict = model_to_save.state_dict()
 
-        # Clean the folder from a previous save
```
Comment (patrickvonplaten) [inline src/diffusers/loaders.py:268]: this was a bad copy-paste and is not needed

**Comment by patrickvonplaten** [comment]
The following code now works without problems:

```py
from diffusers import StableDiffusionPipeline
import torch

pipe = StableDiffusionPipeline.from_pretrained("runwayml/stable-diffusion-v1-5", torch_dtype=torch.float16)
pipe.unet.load_attn_procs("PhanAnh/sd-model-finetuned-lora", weight_name="pytorch_model.bin")

pipe.unet.save_attn_procs("./temp")
pipe.unet.load_attn_procs("./temp")

pipe.unet.save_attn_procs("./temp", safe_serialization=True)
pipe.unet.load_attn_procs("./temp")

pipe.unet.save_attn_procs("./temp", weight_name="lora.bin")
pipe.unet.load_attn_procs("./temp", weight_name="lora.bin")

pipe.unet.save_attn_procs("./temp", weight_name="lora.safetensors", safe_serialization=True)
pipe.unet.load_attn_procs("./temp", weight_name="lora.safetensors")
```

File: src/diffusers/loaders.py [hunk 2]
```diff
@@ -150,13 +150,13 @@ def load_attn_procs(self, pretrained_model_name_or_path_or_dict: Union[str, Dict
 
         model_file = None
         if not isinstance(pretrained_model_name_or_path_or_dict, dict):
-            if (is_safetensors_available() and weight_name is None) or weight_name.endswith(".safetensors"):
-                if weight_name is None:
-                    weight_name = LORA_WEIGHT_NAME_SAFE
+            if (is_safetensors_available() and weight_name is None) or (
+                weight_name is not None and weight_name.endswith(".safetensors")
+            ):
```
Comment (sayakpaul) [inline src/diffusers/loaders.py:155]: Makes sense to add a small comment highlighting that we always look for safetensors first just for easy readability?

File: src/diffusers/loaders.py [hunk 3]
```diff
@@ -245,6 +245,13 @@ def save_attn_procs(
                 need to replace `torch.save` by another method. Can be configured with the environment variable
                 `DIFFUSERS_SAVE_MODE`.
         """
+        weight_name = weight_name or deprecate(
+            "weights_name",
+            "0.18.0",
+            "`weights_name` is deprecated, please use `weight_name` instead.",
+            take_from=kwargs,
+        )
+        print(weight_name)
```
Comment (sayakpaul) [inline src/diffusers/loaders.py:254]: Needs to be removed?

**Review by sayakpaul — APPROVED** [review]
Thanks much!

Left minor nits.

File: src/diffusers/loaders.py (hunk 3 above)
Comment (patrickvonplaten) [inline src/diffusers/loaders.py:254]:
```suggestion
```

File: src/diffusers/loaders.py (hunk 2 above)
Comment (patrickvonplaten) [reply src/diffusers/loaders.py:155]: Good point

File: src/diffusers/loaders.py [hunk 4]
```diff
@@ -150,13 +150,13 @@ def load_attn_procs(self, pretrained_model_name_or_path_or_dict: Union[str, Dict
 
         model_file = None
         if not isinstance(pretrained_model_name_or_path_or_dict, dict):
-            if (is_safetensors_available() and weight_name is None) or weight_name.endswith(".safetensors"):
-                if weight_name is None:
-                    weight_name = LORA_WEIGHT_NAME_SAFE
+            if (is_safetensors_available() and weight_name is None) or (
```
Comment (patrickvonplaten) [inline src/diffusers/loaders.py:153]:
```suggestion
            # Let's first try to load .safetensors weights
            if (is_safetensors_available() and weight_name is None) or (
```

**Pull request merged**